src/finch/tensor.py

```diff
@@ -785,7 +785,25 @@ def _reduce(x: Tensor, fn: Callable, axis, dtype=None):
         result = fn(x._obj, dims=axis)
     else:
         result = fn(x._obj)
-    return Tensor(result)
+
+    if np.isscalar(result):
+        result = jl.Tensor(
+            jl.Element(
+                jc.convert(x.dtype, 0),
+                np.array(result, dtype=jl_dtypes.jl_to_np_dtype[x.dtype])
+            )
+        )
+
+    result = Tensor(result)
+    if dtype != "skip":
```

Comment:
What purpose does `dtype="skip"` serve? I don't think we should ever return Python scalars.

Reply:
These `dtype` rules only apply to `sum` and `prod`. The other reduction functions, `max` etc., skip it.